app/services/orangelight_reindexer.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+class OrangelightReindexer
+  # Reindexes resources indexed into Orangelight by sending record updated messages
+  # for all complete resources.
+  # @param optional logger [Logger] instance of logger class
+  def self.reindex_orangelight(logger: Logger.new(STDOUT))
+    new(logger: logger).reindex_orangelight
+  end
+
+  attr_reader :logger, :ogm_repo_path
+  def initialize(logger:)
+    @logger = logger
+  end
+
+  def reindex_orangelight
+    all_orangelight_resources.each do |resources|
+      resources.each do |resource|
+        begin
+          messenger.record_updated(resource)
+          logger.info("Indexed into Orangelight: #{resource.id}")
+        rescue StandardError => e
+          logger.warn("Error: #{e.message}")
+        end
+      end
+    end
+  end
+
+  private
+
+    def all_orangelight_resources
+      [
+        complete_resources(model: Numismatics::Coin)
+      ]
+    end
+
+    def complete_resources(model:)
+      resources = query_service.find_all_of_model(model: model)
```

Comment:
This is going to be really slow, right? If I spun up a PR real quick to add model filtering and lazy-loaded cursors to `find_by_property` would you use that instead?

Reply:
https://github.com/pulibrary/figgy/pull/3756 has that feature.

Reply:
incorporated